webapp/channels/src/selectors/posts.ts

```diff
@@ -52,3 +58,69 @@ export function isInlineImageVisible(state: GlobalState, postId: string, imageKe
 
     return getGlobalItem(state, StoragePrefixes.INLINE_IMAGE_VISIBLE + currentUserId + '_' + postId + '_' + imageKey, !imageCollapsed);
 }
+
+export function makeCanWrangler() {
+    return createSelector(
+        'makeCanWrangler',
+        getConfig,
+        getCurrentUser,
+        moveThreadsEnabled,
+        (_state: GlobalState, channelType: Channel['type']) => channelType,
+        (_state: GlobalState, _channelType: Channel['type'], replyCount: number) => replyCount,
+        (config: Partial<ClientConfig>, user: UserProfile, enabled: boolean, channelType: Channel['type'], replyCount: number) => {
+            if (!enabled) {
+                return false;
+            }
+            const {
+                WranglerPermittedWranglerRoles,
+                WranglerAllowedEmailDomain,
+                WranglerMoveThreadMaxCount,
+                WranglerMoveThreadFromPrivateChannelEnable,
+                WranglerMoveThreadFromDirectMessageChannelEnable,
+                WranglerMoveThreadFromGroupMessageChannelEnable,
+            } = config;
+
+            let permittedUsers: string[] = [];
+            if (WranglerPermittedWranglerRoles && WranglerPermittedWranglerRoles !== '') {
+                permittedUsers = WranglerPermittedWranglerRoles?.split(',');
+            }
+
+            let allowedEmailDomains: string[] = [];
+            if (WranglerAllowedEmailDomain && WranglerAllowedEmailDomain !== '') {
+                allowedEmailDomains = WranglerAllowedEmailDomain?.split(',') || [];
+            }
+
+            if (permittedUsers.length > 0) {
+                const roles = user.roles.split(' ');
+                const hasRole = roles.some((role) => permittedUsers.includes(role));
+                if (!hasRole) {
```

Comment:
Seems that in the backend we allow for admins at all times besides the permitted users checks, should we also check here `!isAdmin`?